MountList.lua

```diff
@@ -191,6 +191,33 @@ function LM.MountList:PriorityRandom(r)
 end
 
 function LM.MountList:RarityRandom(r)
+    if #self == 0 then return end
+    if not MountsRarityAddon or not MountsRarityAddon.MountsRarity then
+        LM.Debug(" - RarityRandom: MountsRarity not loaded. Defaulting to SimpleRandom.")
+        return self:SimpleRandom(r)
+    end
+
+    local mountsRarity = {}
+    local totalWeight = 0
+
+    for _, m in ipairs(self) do
+        local rarity = MountsRarityAddon.MountsRarity[tostring(m.mountID)]
```

Comment:
I thought about moving this into a `LM.Mount:GetRarity()` function but then we'd have to reference to `MountsRarityAddon.MountsRarity` in two places (because we check for the early return above).

Reply:
I'm not worried about that, I'd prefer to eliminate the memory usage of the mountsRarity table as per the longer comment.

Reply:
If the LibStub used a funtion you could put the or 50 into that.

Alernatively your addon could metatable the data table to return 50 for not-found keys.

Reply:
I changed this to `local rarity = MountsRarity:GetRarityByID(m.mountID) or 50` in the next update.